app/src/main/java/com/pvp/app/ui/screen/goals/GoalScreen.kt

```diff
@@ -0,0 +1,338 @@
+                } else {
+                    " weekly"
+                } + " goal"
+            )
+
+            Spacer(modifier = Modifier.padding(2.dp))
+
+            Text(
+                style = MaterialTheme.typography.labelSmall,
+                fontSize = 10.sp,
+                modifier = Modifier.fillMaxWidth(),
+                textAlign = TextAlign.Center,
+                text = goal.startDate.toString() + " - " + goal.endDate.toString()
+            )
+
+            Spacer(modifier = Modifier.padding(8.dp))
+
+            ProgressBar(goal = goal)
+
+            Spacer(modifier = Modifier.padding(4.dp))
+
+            Row(
+                modifier = Modifier.padding(6.dp)
+            ) {
+                Text(
+                    style = MaterialTheme.typography.bodyMedium,
+                    modifier = Modifier.padding(end = 8.dp),
+                    textAlign = TextAlign.Left,
+                    text = "Set goal: ${
+                        when (goal.steps) {
+                            true -> "${goal.goal.toInt()} steps"
+                            false -> "${goal.goal} km"
+                        }
+                    }"
+                )
+
+                Icon(
+                    imageVector = goal.activity.icon,
+                    contentDescription = "Activity icon",
+                )
+            }
+
+            if (goal.steps) {
+                Spacer(modifier = Modifier.padding(2.dp))
+
+                Text(
+                    style = MaterialTheme.typography.bodyMedium,
+                    modifier = Modifier.padding(start = 6.dp),
+                    textAlign = TextAlign.Left,
+                    text = "Your average " + when (goal.monthly) {
+                        true -> "monthly steps: $monthSteps"
+                        false -> "weekly steps: ${monthSteps / 30 * 7}"
+                    }
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun GoalTypeFilter(
+    filter: GoalFilter,
+    onClick: (GoalFilter) -> Unit
+) {
+    Row(
+        horizontalArrangement = Arrangement.SpaceBetween,
+        modifier = Modifier
+            .background(
+                MaterialTheme.colorScheme.surfaceContainer,
+                MaterialTheme.shapes.medium
+            )
+            .fillMaxWidth()
+            .padding(vertical = 4.dp)
+    ) {
+        GoalFilter.entries.forEach { filterNew ->
+            GoalTypeSelector(
+                filter = filterNew,
+                isSelected = filter == filterNew,
+                onClick = { onClick(filterNew) },
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth()
+                    .height(40.dp)
+            )
+        }
+    }
+}
+
+@Composable
+private fun GoalTypeSelector(
+    filter: GoalFilter,
+    isSelected: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Box(contentAlignment = Alignment.Center,
```

Comment:
Put in new line; Parameters closing bracket will have to be shifted into new line too.